### PR TITLE
Release candidate+lwip 2.1.2

### PR DESCRIPTION
**Pull request by dan4thewin**
This pulls in the commits from lwIP 2.1.2 PRs against master.

### CODE REVIEW

File: libraries/abstractions/platform/test/iot_test_platform_threads.c [hunk 1]
```diff
@@ -71,7 +71,9 @@ TEST_TEAR_DOWN( UTIL_Platform_Threads )
 TEST_GROUP_RUNNER( UTIL_Platform_Threads )
 {
     RUN_TEST_CASE( UTIL_Platform_Threads, IotThreads_CreateDetachedThread );
-    RUN_TEST_CASE( UTIL_Platform_Threads, IotThreads_ThreadPriority );
+    #if ( configUSE_TRACE_FACILITY == 1 )
```
Comment (dcgaws) [inline libraries/abstractions/platform/test/iot_test_platform_threads.c]: Is this related to the lwIP upgrade?

Comment (dan4thewin) [reply libraries/abstractions/platform/test/iot_test_platform_threads.c]: Yes.  The MediaTek board is quite constrained.  Disabling the trace facility permitted a successful compilation.
